sql/rowexec/ddl_iters.go

```diff
@@ -543,6 +553,18 @@ func (i *modifyColumnIter) rewriteTable(ctx *sql.Context, rwt sql.RewritableTabl
 			return false, err
 		}
 
+		// remap old enum values to new enum values
+		if isOldEnum && isNewEnum {
+			oldIdx := int(newRow[newColIdx].(uint16))
+			oldStr, _ := oldEnum.At(oldIdx)
+			newIdx := newEnum.IndexOf(oldStr)
+			if newIdx == -1 {
+				// TODO: convert to truncated warning, and somehow still show old enum value
```

Comment:
I did a super quick check with MySQL's behavior and I saw an error when I tried to alter the enum to remove a value that was in use. seems like that matches the behavior you have here: an error is returned and the data and column definition do not change. 

So, seems like you may not need this TODO after all? My testing was on MySQL 8.4 and was pretty quick, so I may have just missed something.

Reply:
I tested this in `8.0.29`, so they must've changed this behvior.
Thanks for checking! Will remove the TODO